packages/web/src/ui/views/viewer.js

```diff
@@ -63,39 +63,43 @@ const viewer = (state, i18n) => {
     render = prepareRender(viewerOptions)
     const gestures = require('most-gestures').pointerGestures(el)
 
+    let delta = [0,0];
+    let deltaPan = [0,0];
+    let timer;
+
     // rotate
     gestures.drags
       .forEach((data) => {
         const ev = data.originalEvents[0]
-        const shiftKey = (ev.shiftKey === true) || (ev.touches && ev.touches.length > 2)
-        if (!shiftKey) {
-          const now = Date.now()
-          const ms = now - lastRotate
-          if (ms > (1000 / rotateRate)) {
-            const delta = [data.delta.x, data.delta.y].map((d) => -d)
-            const updated = orbitControls.rotate({ controls, camera, speed: rotateSpeed }, delta)
-            controls = { ...controls, ...updated.controls }
-            lastRotate = now
-          }
-        }
-      })
-    // pan
-    gestures.drags
-      .forEach((data) => {
-        const ev = data.originalEvents[0]
+        const {x,y} = data.delta;
         const shiftKey = (ev.shiftKey === true) || (ev.touches && ev.touches.length > 2)
         if (shiftKey) {
-          const now = Date.now()
-          const ms = now - lastPan
-          if (ms > (1000 / panRate)) {
-            const delta = [data.delta.x, data.delta.y].map((d) => d)
-            const updated = orbitControls.pan({ controls, camera, speed: panSpeed }, delta)
-            camera.position = updated.camera.position
-            camera.target = updated.camera.target
-            lastPan = now
-          }
+          deltaPan[0] += x;
+          deltaPan[1] += y;          
+        }else{
+          delta[0] -= x;
+          delta[1] -= y;
         }
-      })
+        if(!timer) timer = requestAnimationFrame(doRotatePan);
+    })
+
+    const doRotatePan = (timestamp) => {
+      timer = null;
+      if(delta[0] || delta[1]){      
+        const updated = orbitControls.rotate({ controls, camera, speed: rotateSpeed }, delta)
+        delta = [0,0];
+        controls = { ...controls, ...updated.controls }
+      }
+      
+      if(deltaPan[0] || deltaPan[1]){      
+        const updated = orbitControls.pan({ controls, camera, speed: panSpeed }, deltaPan)
+        deltaPan = [0,0];
+        camera.position = updated.camera.position
+        camera.target = updated.camera.target
+      }
+
+      updateAndRender(timestamp)
```

Comment:
Hmm... there may be some issues here, as updateAndRender() can be heavy if there are lots of polygons.

Reply:
well I have just puched draft commit without throttling based on ms,
and changed zoom to use new concept, and examples\core\primitives\primitives3D works much smoother on my phone now than on current live verzion jscad.xyz.

for this change to be final I would need to clean-up unused variables.

further throttling options could be considered, but browser will not force animation at 60 FPS r whatever if it is not able to catchup.

Reply:
I have tried reversing the logic (did not commit/push this experiment):

 - zoom/pan/roate events are collected only without calling update for next animation frame
 - animation frames are used but throttled (current default 10 FPS)
 - inside animation frame both doRotatePanZoom and updateAndRender are called
   (calling doRotatePanZoom is ok as it will do nothing if no events have been collected)

this throttling did not help at all in situation with too many stuff to render 
(on my phone:  "Show grid" plus "3d primitives example" is enough to make it slow)

when poly count was manageable throttling only made it all look less performant.

Reply:
I have noticed that app works much better in landscape.
on my phone resize function is messed up.
so canvas is made too big when in portrait mode.

I will look into this further but will split that into a separate pull request.

Now comparing the performance of new code vs jscad.xyz in landscape ... new code is much smoother.